### DISCUSSION TITLE
Questions/issues with new cursor highlighting during editing

### DISCUSSION THREAD

**Discussion by JaniceManwiller** [GitbookIO/community#356] [0. General bugs and questions] [answered]
I noticed today that when I am making edits in a change request, I see the bright cursor with my name. It almost looks like GitBook thinks that another user is making the changes.

I gather from the change log that this is intended to be a collaboration feature, but:

- This is a published space. The changelog said that the colored cursors were only used in unpublished spaces.
- The space uses GitSync. The changelog entry said that it does not happen where GitSync is enabled.
- I am the only user in the change request. I don't have GitBook open in another tab or anything like that.

Another random issue is that I can no longer use the right arrow key to navigate. Up, down, and left work fine. I would say it was my keyboard, but I don't have the same issue in other tools.

**Comment by SamyPesse** [comment]
Hi @JaniceManwiller, We'll investigate the bug as it's definitely not expected to see your own cursor as a real-time participant.

Is it still happening after you refresh the tab? Are you observing it on tabs that have been inactive for a while?

>Another random issue is that I can no longer use the right arrow key to navigate. Up, down, and left work fine. I would say it was my keyboard, but I don't have the same issue in other tools.

It looks like a side effect of the the bug, where the other cursor is blocking the use of right arrow.

**Reply by JaniceManwiller** [reply]
I'm actually not seeing it right now, after closing and reopening the tab a second time. After the first time it was still happening.

**Comment by dfinster** [comment]
I am seeing the same behavior this morning. It has happened twice this morning so far. 

* I saw it as soon as I started working, launching Chrome after a fresh reboot
* Again, a couple minutes after I reloaded the tab in frustration

Once is a glitch, twice is a bug, so I stopped trying to work and came here to find out if I'm the only one. Editing an article with this bug is extremely difficult because I have the same issue:

>  I can no longer use the right arrow key to navigate. Up, down, and left work fine.

Maybe related: Sometimes reloading the tab shows my profile pic in the top bar, as if I had two tabs open, and sometimes it doesn't. I do not have two tabs open for GitBook. 

If I could make a comment about this feature in general:
_We are not looking for a collaboration suite like Google Docs. We just want to publish our documentation to the Internet. I'm not sure what problem GitBook is trying to solve with real-time collaboration. Regardless, the editing experience for GitBook has been so poor in my 6-month's experience using it that I typically do all my drafts and review cycles in other tools and try to use GitBook as little as possible, only for the final markup and publication._

**Reply by JaniceManwiller** [reply]
This is also how I use GitBook - to publish our end-user documentation. It's not a collaborative tool for us either. All draft content is created elsewhere and then ported into GitBook CRs when it's time to publish the new or updated content. The only collaboration is to have SMEs review the CRs.

**Reply by dfinster** [reply]
@SamyPesse I'd be interested to know if Enterprise users, like my company, can choose to be on a slower release cycle with planned releases and release notes. We don't need to be in this constant beta test with new features and behavior appearing without notice.

**Reply by SamyPesse** [reply, marked as answer]
Hi @JaniceManwiller and @dfinster, a fix was released yesterday evening (8 PM CET). Let me know if you still notice the issue (after a hard refresh of your page to avoid any cache).

> I'd be interested to know if Enterprise users, like my company, can choose to be on a slower release cycle with planned releases and release notes. We don't need to be in this constant beta test with new features and behavior appearing without notice.

We can't do this at the moment as the application is a unique one for all customers, with a unique release cycle.
Also in a case like this one, this bug was a regression on a bug fix that was released a few hours earlier, it was not caused by new features/experiments. For new features and experiments, we are now using feature flagging to avoid impacting our customers and test them on a community of beta users.

**Reply by dfinster** [reply]
@SamyPesse Thanks for the fix and the feature flag information! I'll pass that along.

**Comment by dfinster** [comment]
This afternoon, I was able to trigger this bug again by clicking the left menu to change pages while editing a CR. It's also worth noting that changing pages today is quite slow. I'm greeted with the "wait-spinner" for many seconds when navigating to a new page I haven't visited yet since loading the browser. I don't recall it being this painful to navigate before recently.
